File: classes/python/PrivateByClosure.py
#! /usr/bin/env python

class A:
    def __init__(self):
        counter = 0
        def tick():
            nonlocal counter
            counter += 1
        self.tick = tick
        def count():
            return counter
        self.counter = count

a = A()
a.tick()
a.tick()
a.tick()
print(a.counter())

File: classes/javascript/PrivateByClosure.js
#! /usr/bin/env node
"use strict";

class A {
  constructor(value){
    let counter = 0;
    this.tick = () => counter += 1;
    this.count = () => counter;
  }
}

const a = new A();
a.tick();
a.tick();
a.tick();
console.log(a.count());
